larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArFormattingHelper.cc
+ *
+ *  @brief  Implementation of the formatting helper class.
+ *
+ *  $Log: $
+ */
+
+#include <iomanip>
+
+#include "Pandora/StatusCodes.h"
+
+#include "larpandoracontent/LArHelpers/LArFormattingHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+    void LArFormattingHelper::SetStyle(const LArFormattingHelper::Style &style, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(style), stream);
+    }
+   
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::SetColour(const LArFormattingHelper::Colour &colour, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(colour), stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetStyle(std::ostream &stream)
+    {
+        LArFormattingHelper::SetStyle(REGULAR, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetColour(std::ostream &stream)
+    {
+        LArFormattingHelper::SetColour(DEFAULT, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::Reset(std::ostream &stream)
+    {
+        LArFormattingHelper::ResetColour(stream);
+        LArFormattingHelper::ResetStyle(stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintFormatCharacter(const unsigned int &code, std::ostream &stream)
+    {
+        stream << LArFormattingHelper::GetFormatCharacter(code);
```

Comment:
Can this operation fail? Doesn't the operator return something you can check, then raise an exception?

Reply:
I can check if stream::fail() returns true!